server.go

```diff
@@ -177,6 +177,99 @@ func (ps *peerState) forAllPeers(closure func(sp *serverPeer)) {
 	ps.forAllOutboundPeers(closure)
 }
 
+// ResolveNetworkAddress returns the network address string representation
+// (ip:port) of the node based on cummulated suggestions from outbound peers.
+func (ps *peerState) ResolveNetworkAddress() string {
+	peers := ps.outboundPeers
+
+	// Return if there are no outbound peers or only a single outbound peer.
+	if len(peers) <= 1 {
+		return ""
+	}
+
+	suggestions := make(map[string]uint32, len(peers))
+	for _, p := range peers {
+		if p.peerNetworkAddress == "" {
```

Comment:
This is all incredibly racy.  This field is being access unprotected from multiple goroutines.